loadtest/control/gencontroller/actions.go

```diff
@@ -368,3 +411,33 @@ func (c *GenController) joinTeam(u user.User) control.UserActionResponse {
 
 	return control.UserActionResponse{Info: fmt.Sprintf("joined team %s", team.Id)}
 }
+
+// chooseChannel will pick a channelID randomly from the range of indexes.
+// If the chosen channelID has exceeded the number of channelmembers, it will
+// select another one in the range until it has found one.
+func chooseChannel(dist ChannelsDistribution, u user.User) (string, error) {
+	minIndex := int(dist.MinIndexRange * float64(len(st.channels)))
+	maxIndex := int(dist.MaxIndexRange * float64(len(st.channels)))
+
+	if maxIndex-minIndex <= 1 {
+		return "", errors.New("not enough channels to select from; either increase range or increase number of channels to create")
+	}
+
+	var channelID string
+	for {
+		target := rand.Intn(maxIndex-minIndex) + minIndex
+		if target >= maxIndex {
+			target = maxIndex - 1
+		}
+		channelID = st.channels[target]
+
+		members, err := u.Store().ChannelMembers(channelID)
+		if err != nil {
+			return "", err
+		}
+		if len(members) > int(dist.MemberLimit) {
+			continue
+		}
```

Comment:
Can we enter an endless loop here at some point? Maybe we can add an upper bound in the number of tries.